AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/hierarchical-state-machine-swiftTests/StateMachineTests.swift

```diff
@@ -156,8 +140,9 @@ class StateMachineTests: XCTestCase {
             eventType: .incrementAndDoActions([action1])
         )
 
-        testMachine.send(event)
-        wait(for: [action1WasExecuted, action2WasExecuted], timeout: 0.1, enforceOrder: true)
```

Comment:
Note: I could not figure out a way to assert the order of expectation in async test context.

Reply:
We do need to test this. I'm assuming there's something about the async context that is causing the `wait(for:,timeout:,enforceOrder:)` method to not work, so maybe we have assertions built into task 2 that ensures that task 1 has already been invoked? something like:

```swift
let action1WasExecuted = AtomicValue(initialValue: false)
let action2WasExecuted = AtomicValue(initialValue: false)

let action1 = BasicAction(identifier: "basic") { dispatcher, _ in
    action1WasExecuted.toggle()
    let action2 = BasicAction(identifier: "basic") { _, _ in
        XCTAssert(action1WasExecuted.value)
        action2WasExecuted.toggle()
    }
    ...
}

await testMachine.send(event)
XCTAssert(action1WasExecuted.value)
XCTAssert(action2WasExecuted.value)
```

Reply:
Thank you, added a variation of this.